### DISCUSSION TITLE
How should I compile different libs for different instruction sets ?

### DISCUSSION THREAD

**Discussion by dreamcmi** [xpack-dev-tools/riscv-none-elf-gcc-xpack#6] [Q&A] [answered]
I'm compiling the riscv toolchain using the build script in the repository, and in the generated binaries, I found that there are only some bases a library and ldscripts folder in the `xpack-riscv-none-elf-gcc-12.2.0-1\riscv-none-elf\lib` directory , but the same directory under github release will have subfolders with different instruction sets, such as `rv32e`, `rv32i`, `rv32ima`. How should I compile to output these files? 
My build command is `bash scripts/helper/build.sh --develop --win64` and the branch used `xpack-develop`.
![git](https://user-images.githubusercontent.com/54615441/197490499-3d1c2c7a-4f7f-4a5b-929b-55bafc69eb61.png)
![selfbuild](https://user-images.githubusercontent.com/54615441/197490513-7ee658db-0dc3-45e3-927e-ddec1f99ecb3.png)

**Comment by ilg-ul** [comment, marked as answer]
The subfolders with different instruction sets are the result of the _multilib_ configuration. For one reason or another, your build seems to run without multilibs. Anyway, if you enable `--develop`, this restricts the number of multilibs, to save some compile time, which may go up to several hours.

BTW, the `xpack-develop` branch may not build properly occasionally.

**Reply by dreamcmi** [reply]
Thanks, I tried switching the branch to `xpack` and canceled `--develop`, then it compiled successfully after about 4h of compilation. But one more question, can I specify to compile only certain libs?

**Reply by ilg-ul** [reply]
> can I specify to compile only certain libs?

Yes, edit the list of multilibs.

**Reply by dreamcmi** [reply]
Thanks, I found the configuration of multilab in `common-versions-source.sh`, but the comment on line 81 is in doubt.      
```shell 
# DO NOT add the combination that is already given as the default!
 # rv32imac-ilp32-- \ 
```
Does gcc12 not support rv32imac? Or does rv32imac do not need to compile lib independently?

**Reply by ilg-ul** [reply]
The `rv32imac` is passed as the default architecture, you don't have to pass it again.

**Reply by dreamcmi** [reply]
Thanks a lot.
